### PR TITLE
log-backup: adapt user-defined TLS config

### LINKED ISSUE #12668: log-bakcup: support custom TLS
## Development Task
Support TLS config in the [security] field for the gRPC connection to PD used in PiTR.

### PR DESCRIPTION
**Pull request by YuJuncen**
Signed-off-by: Yu Juncen <yujuncen@pingcap.com>

### What is changed and how it works?

Issue Number: Close #12668

What's Changed:
- Added a new metastore type `LazyEtcd`, which supports connecting with TLS.
- Modify the `Endpoint::new`, now it receives `MetaStore` instead of `endpoints`.

```commit-message
log-backup: adapt user-defined TLS config
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
I have tested it by:
1. Create a cluster by `tiup`.
2. Enable TLS by `tiup cluster tls <cluster-name> enable`.
3. Run log backup, there should not be errors or panic.
- No code

### Release note 

```release-note
Fix a bug that caused PiTR cannot work if some of `security.{ca-path, cert-path, key-path}` configured.
```

### CODE REVIEW

File: components/backup-stream/src/metadata/store/lazy_etcd.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        if let Some(tls) = &self.tls {
+            opts = opts.with_tls(tls.clone())
+        }
+        opts = opts.with_keep_alive(self.keep_alive_interval, self.keep_alive_timeout);
+        opts
+    }
+}
+
+impl LazyEtcdClient {
+    pub fn new(endpoints: &[impl ToString], conf: ConnectionConfig) -> Self {
```
Comment (kennytm) [inline components/backup-stream/src/metadata/store/lazy_etcd.rs:35]: why can't we just use `&[String]` here 😅

File: components/backup-stream/src/endpoint.rs [hunk 2]
```diff
@@ -632,6 +551,7 @@ where
             .await
             .map_err(|err| Error::from(err).report("failed to get tso from pd"))
             .unwrap_or_default();
+        cm.update_max_ts(pd_tso);
```
Comment (3pointer) [inline components/backup-stream/src/endpoint.rs:554]: why move this line?

Comment (YuJuncen) [reply components/backup-stream/src/endpoint.rs:554]: we should update this to pd ts before we trying to resolve, so there won't be new async-commit prewrite uses [0, pd_ts] to commit during we resolving.

**Review by 3pointer — APPROVED** [review]
LGTM

File: components/backup-stream/src/endpoint.rs [hunk 3]
```diff
@@ -444,24 +358,30 @@ where
...
+            if kvs.as_ref().map(|x| x.is_empty()).unwrap_or(true) {
+                return;
+            }
+            let kvs = kvs.unwrap();
+
             HANDLE_EVENT_DURATION_HISTOGRAM
-                .with_label_values(&["get_router_lock"])
-                .observe(sw.lap().as_secs_f64());
+            .with_label_values(&["to_stream_event"])
+            .observe(sw.lap().as_secs_f64());
```
Comment (kennytm) [inline components/backup-stream/src/endpoint.rs:384]: strange de-indentation

**Review by kennytm — APPROVED** [review]
rest lgtm

**Pull request merged**